### DISCUSSION TITLE
Merge Queue: Allow specifying separate status checks for the merge queue

### DISCUSSION THREAD

**Discussion by MythicManiac** [community/community#47548] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

We've been trying out the new merge queue feature, and discovered one of our required external status checks (https://cla-assistant.io/) never runs for the merge queue. Sample screenshot below:
![image](https://user-images.githubusercontent.com/8225825/219025390-168cc9c2-ee63-4727-82d5-0415387d1054.png)

It would be useful if there was a way to define which status checks are required for a merge queue, and which are for PRs only. We could easily drop several of the status checks here (such as publishing), as those are already ran either pre-merge or post-merge.

**Comment by unknowVariable** [comment]
Hi there,

Thank you for your feedback regarding the merge queue feature on GitHub. I understand that you would like to have more control over which status checks are required for the merge queue and which are only required for pull requests.

Currently, the merge queue feature requires all required status checks to pass before a pull request can be merged, regardless of whether the check is for the merge queue or for the pull request. However, I can understand the benefit of being able to differentiate between the two types of status checks and will pass along your feedback to the GitHub team for consideration.

In the meantime, one possible workaround would be to modify your repository's required status checks to include only those checks that are necessary for the merge queue and move the other checks to the optional status checks section. This way, the merge queue will only require the necessary checks, and the remaining checks will still be required for pull requests.

I hope this helps, and thank you again for your feedback.

**Comment by widhalmt** [comment]
Hi,

I was just going to give the same feedback. We have a lot different checks that target different scenarios. We're deveolping Ansible Collections and most checks are only useful when the targeted code has been changed. But before finally merging into `main` we want to run a check for the whole collection to see if all moving parts still work together. These checks can require significant ressources so we don't want to run them on every single push, only when we think the PR is ready for merge. So it looks like we also need a check that's only triggered when the PR is added to a queue. It should be neccessary before leaving the queue but not to enter the queue.

Maybe this helps to give an extra point of view.

**Comment by lars-reimann** [comment]
I'd also like separate required checks for pull requests and merge queues. 

### Use case 1: Publishing documentation to `readthedocs.org`

I've configured Read the Docs to also build documentation for pull requests to ensure that nothing is broken. That's also why I would like to make this check required for pull requests. However, inside the merge queue, I don't even know how I could tell Read the Docs to build the documentation again.

### Use case 2: Format checks

I want to verify that PR titles follow the [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) spec, so tools like `semantic-release` work. This check should also be required for pull requests. But inside the merge queue, the commit message is not going to change, so it's pointless to run the check again.

### Conclusion

I want to specify a set of conditions that must be true before a pull request is allowed to enter the merge queue. Additionally, I want to specify another set of conditions that gets verified and enforced when the pull request reaches the front of the merge queue.

**Comment by dsotirakis** [comment]
Same for us, using Drone CI.

Unfortunately, Drone server doesn't yet accept the `enqueued` action from the payload, thus it waits the merge queue build to be treated as a `pull_request` event with an `opened` or a `synchronized` action, which is not the case.

We are able to bypass this issue by triggering builds to run on `gh-readonly-queue/**` triggered on push events, but that's not working because it expects the `pull_request` event to run, because it's specified in the branch rules.

What we would like is to be able to set merge queues checks separately from the pull requests checks, so the merge queues are agnostic to the CI tool everyone uses.

**Comment by gilvan-fernandes** [comment]
Up. Same issue.

**Comment by dahlbyk** [comment]
Bump. My specific scenario is that I would like a long-running verification to always run against the PR, but only be _required_ to get past the MQ.
